mobius-rx/src/main/java/com/spotify/mobius/rx/Transformers.java

```diff
@@ -119,4 +121,57 @@ public void call(F f) {
       }
     };
   }
+
+  /**
+   * Creates an {@link Observable.Transformer} that will flatten the provided {@link Func1} into the
+   * stream as an {@link Observable} every time it receives an effect from the upstream effects
+   * observable. This will result in calling the function on the specified scheduler, and passing it
+   * the requested effect object then emitting its returned value.
+   *
+   * @param function the {@link Func1} to be invoked every time the effect is requested
+   * @param scheduler the {@link Scheduler} to be used when invoking the function
+   * @param <F> the type of Effect this transformer handles
+   * @param <E> the type of Event this transformer emits
+   * @return an {@link Observable.Transformer} that can be used with a {@link
+   *     SubtypeEffectHandlerBuilder}.
+   */
+  static <F, E> Observable.Transformer<F, E> fromFunction(
+      final Func1<F, E> function, @Nullable final Scheduler scheduler) {
+    return new Observable.Transformer<F, E>() {
+      @Override
+      public Observable<E> call(Observable<F> effectStream) {
+        return effectStream.flatMap(
+            new Func1<F, Observable<E>>() {
+              @Override
+              public Observable<E> call(final F f) {
+                Observable<E> eventObservable =
+                    Observable.fromCallable(
+                        new Callable<E>() {
+                          @Override
+                          public E call() throws Exception {
+                            return function.call(f);
+                          }
+                        });
+                return scheduler == null ? eventObservable : eventObservable.subscribeOn(scheduler);
```

Comment:
Wouldn't it be better to pass the scheduler to flatMap instead?

Reply:
AFAIK this isn't possible. None of the Flatmap variants have a `Scheduler` as a parameter. Do you know something I don't? :)